Examples.md

```diff
@@ -166,6 +201,17 @@ server.start()
 await server.wait_for_termination()
 ```
 
+### Option 3
+
+```Python
+server = grpc.server()
+server.add_insecure_port(':50051')
+helloworld_pb2_grpc.add_AsyncGreeterServicer_to_server(Greeter(), server)
```

Comment:
I can understand the value of async stub, but what is the benefit of async servicer? We can support sync / async handler in the same servicer.

Reply:
Can you clarify, please? Do you suggest to introduce some proto extension to mark individual method within a service as sync or async?

Reply:
> In the short term, we want the generated gRPC code to be one file, and can be used for both Python 2 and Python 3.

Why not generate async stubs in separate py3-only file? What will it break?

> In that case, the typing info isn't included in the generated file

Mypy for both Python 2 and Python 3 supports typing via `# type: ` comment, so either of methods can be typed.
Also, either of them supports putting python3-style typing into separate `.pyi` file.

It feels complicated to allow both `StreamingHi` and `AsyncStreamingHi` within single servicer implementation:
1. How would typing work?
2. How would grpc server determine which one to call?

Reply:
I misinterpret your first comment. I thought the snippet is what you are suggesting. What is your suggestion about servicer interface design? How can we help users migration from existing API to async API?

If we want to introduce strong typing, sure, the separate Python 3 only file is the only way. I will try to add a section in the gRFC.

Reply:
I have an example of server's servicer stub in the pull request:
https://github.com/lidizheng/grpc-api-examples/blob/71955b583783be2f1f90233a07acb6a00b6a6632/Examples.md#option-2-2

Client should be symmetric, though `Unified Stub Call` can somehow complicate it (but let's discuss it in a separate thread).

> If we want to introduce strong typing, sure, the separate Python 3 only file is the only way. I will try to add a section in the gRFC.

We can either generate separate `.pyi` files or put special comments with `# type: ` prefix, either will work.